ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -179,12 +192,19 @@ proc new*(_: type JsonRpcSubscriptions,
         subscriptions.subscriptionMapping[originalId] = newId
         return await getChanges(originalId)
       else:
-        raise e
+        raise newException(SubscriptionError, "HTTP polling: There was an exception while getting subscription changes: " & e.msg, e)
```

Comment:
🟡  I'm wondering now if maybe `poll` and `getChanges` should return Results instead of raising or re-raising to make things more readable. They should still be able to raise `CancelledError` though. Maybe not, just a thought I had while thinking about adding in the ability to get logs for events that happen between losing a subscription and re-subscribing.

Reply:
Not sure what you mean. Yes, we throw an exception here, but this is not propagated above `poll` call site, where it is caught and then returned as a failed result into the callback...

https://github.com/codex-storage/nim-ethers/pull/91/files/3c64903822b15c7de2d6620a3ffbea0e879fc772#diff-597f9431d9147b024d3410322caa9432b95c8543f1c21c1c6ed784c9704fa943R203

Does that address what are you suggesting here?

Reply:
I was referring to having `getChanges` and `poll` return Results instead of bubbling exceptions which ultimately will get passed to callback:
```diff
- proc getChanges(id: JsonNode): Future[JsonNode] {.async: (raises: [CancelledError]).} =
+ proc getChanges(id: JsonNode): Future[?!JsonNode] {.async: (raises: [CancelledError]).} =
```
and
```diff
- proc poll(id: JsonNode) {.async: (raises: [CancelledError]).} =
+ proc poll(id: JsonNode): Future[?!void] {.async: (raises: [CancelledError]).} =
```

Reply:
There are two `poll` functions, I was referring to the one that does not call the callbacks.

Reply:
Ah I see, looking at the code again, ignore that last comment 👍